app/src/androidTest/java/com/alexzh/testapp/ui/settings/SettingsActivityTest.kt

```diff
@@ -0,0 +1,220 @@
+     *  and "Layout Inspector" can help with it.
+     *  - the checking descendant property can done with RecyclerViewMatchers#atPosition ViewMatcher
+     *  (can be not implemented yet).
+     *  - the preference item often has 2 descends: title and value.
+     */
+    @Test
+    fun shouldChangeDefaultValueOfDefaultReplyAction() {
+        fail()
+    }
+
+    /**
+     * Test case:
+     *  - the "Sync email periodically" has "false" default value
+     *  - the "Sync email periodically" has "true" value after changing property
+     *
+     * Notes:
+     *  - the "Sync email periodically" string can be found in stings.xml file as sync_title const
+     *  (R.string.sync_title)
+     *
+     * Hint(s):
+     *  - check which preferences are used "Support Libraries" vs "AndroidX" (read more about
+     *  difference in comment for the class). You can check the /res/xml/root_preferences.xml
+     *  for understanding which preferences are used in the project.
+     *  - check ID of Preferences container (ListView or RecyclerView). The "UiAutomatorViewer"
+     *  and "Layout Inspector" can help with it.
+     *  - the checking descendant property can done with RecyclerViewMatchers#atPosition ViewMatcher
+     *  (can be not implemented yet).
+     *  - the preference item often has 2 descends: title and value.
+     */
+    @Test
+    fun shouldChangeDefaultValueOfSyncEmailPeriodically() {
+        fail()
+    }
+
+    /**
+     * Test case:
+     *  - the "Download incoming attachments" is disabled
+     *  - the "Download incoming attachments" has "false" default value
+     *  - the "Sync email periodically" has "true" value after changing property
+     *  - the "Download incoming attachments" enabled after changing "Sync email periodically" to "true"
+     *  - the "Download incoming attachments" change value after changing property
+     *
+     * Notes:
+     *  - the "Download incoming attachments" string can be found in stings.xml file as attachment_title const
+     *  (R.string.attachment_title)
+     *  - the "Sync email periodically" string can be found in stings.xml file as sync_title const
+     *  (R.string.sync_title)
+     *
+     * Hint(s):
+     *  - check which preferences are used "Support Libraries" vs "AndroidX" (read more about
+     *  difference in comment for the class). You can check the /res/xml/root_preferences.xml
+     *  for understanding which preferences are used in the project.
+     *  - check ID of Preferences container (ListView or RecyclerView). The "UiAutomatorViewer"
+     *  and "Layout Inspector" can help with it.
+     *  - the checking descendant property can done with RecyclerViewMatchers#atPosition ViewMatcher
+     *  (can be not implemented yet).
+     *  - the preference item often has 2 descends: title and value.
+     */
+    @Test
+    fun shouldChangeDefaultValueOfDownloadIncomingAttachments() {
```

Comment:
I am wondering if there are too many cases in this file. Maybe you should consider only two things for settings. One in a primary category another maybe somewhere deep down so it is harder to get to it in the test? I am remembering the workshop here in Lodz and I think we skipped a few.

Or split the exercises into workshop part and home work part?

Reply:
During the workshop, all test cases will be marked as part of the workshop and home task. I'll create a separate branch for the workshop.